android/java/org/chromium/chrome/browser/search_engines/settings/BraveSearchEngineAdapter.java

```diff
@@ -89,12 +99,15 @@ static public String getDSEShortName(Profile profile, boolean readJavaPrefOnly)
     }
 
     static public TemplateUrl getTemplateUrlByShortName(Profile profile, String name) {
-        List<TemplateUrl> templateUrls =
-                BraveTemplateUrlServiceFactory.getForProfile(profile).getTemplateUrls();
-        for (int index = 0; index < templateUrls.size(); ++index) {
-            TemplateUrl templateUrl = templateUrls.get(index);
-            if (templateUrl.getShortName().equals(name)) {
-                return templateUrl;
+        TemplateUrlService templateUrlService =
+                BraveTemplateUrlServiceFactory.getForProfile(profile);
+        if (templateUrlService != null) {
```

Comment:
why do we check on `isLoaded` on some places and not on others?

Reply:
The other `isLoaded` is only used in `start()` function in the case of an OTR profile, when we are likely creating the profile and the service initialization can be slower than the preference is opened.